storage/boltz/migration.go

```diff
@@ -0,0 +1,75 @@
+package boltz
+
+import (
+	"github.com/michaelquigley/pfxlog"
+	"github.com/netfoundry/ziti-foundation/util/errorz"
+	"go.etcd.io/bbolt"
+)
+
+type MigrationStep struct {
+	errorz.ErrorHolderImpl
+	Component      string
+	Ctx            MutateContext
+	CurrentVersion int
+}
+
+type Migrator func(step *MigrationStep) int
+
+type MigrationManager interface {
+	Migrate(component string, migrator Migrator) error
+}
+
+func NewMigratorManager(db Db) MigrationManager {
+	migrator := &migrationManager{
+		db: db,
+	}
+	return migrator
+}
+
+type migrationManager struct {
+	db Db
+}
+
+func (m *migrationManager) Migrate(component string, migrator Migrator) error {
+	return m.db.Update(func(tx *bbolt.Tx) error {
```

Comment:
Is it the migration manager's job to make a backup and restore if any of the migrations error?

Reply:
I haven't added backup yet. I think adding a Snapshot method to Db makes sense, which can then be invoked from the migrator.

Reply:
Migrations are done in a transaction, so if anything fails, it will rollback. If someone wants to rollback b/c things aren't working, they will need to copy the file themselves.